### PR TITLE
Refactor package

### PR DESCRIPTION
**Pull request by Americas**
- Move code to src
- Move auxiliary methods to an utils file
- Remove bluebird as a dependency
- Remove lodash as a dependency
- Bump jest to latest version (tests were also updated since there were some strange interactions that would exit the tests before completion)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -95,23 +71,28 @@ class ProcessManager {
...
+          const error = await promises[i];
+
+          if (error) {
+            this.errors.push(error);
+          }
         }
-      })
-      .then(errors => this.errors.push(..._.compact(errors)));
+      }
+    });
```
Comment (kurayama) [inline src/index.js:95]: I'm not a fan of using util.inspect to check the status of a promise.
Can't we move the race inside the promises map and do a allSettled instead?

```js
  hook(type, ...args) {
    const hooks = this.hooks.filter(hook => hook.type === type);
    const promises = hooks.map(({ handler }) => Promise.race([utils.reflect(handler, args), utils.timeout(this.timeout)]));

    return Promise.allSettled(promises)
    .then(async () => {
      for (let i = 0; i < hooks.length; ++i) {
        if (promises[i].status === 'rejected') {
          this.errors.push(hooks[i].timeoutError);

          log.info(`Timeout: ${hooks[i].name} took too long to complete ${type} hook`);
        } else if (promises[i].value) {
          this.errors.push(promises[i].value);
        }
      }
    });
  }
```

And add the util:

```js
/**
 * Creates promise that will reject after the given time (in ms).
 */

module.exports.timeout = function(ms) {
  return new Promise((resolve, reject) => setTimeout(reject, ms));
};
```

**Review by kurayama — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (Americas) [reply src/index.js:95]: I was avoiding bumping the minimum node engine :smile:

Comment (kurayama) [reply src/index.js:95]: The CI should reflect that then.
Run a matrix with multiple node versions.

Promise.allSettled is node 12.9+, should be widespread by now.

**Comment by kurayama** [comment]
On .npmignore, the `!index.js` pattern must be changed to ` !src/**`

**Review by kurayama — APPROVED** [review]

**Pull request merged**